src/clojure/nrepl/middleware/print.clj

```diff
@@ -184,6 +143,12 @@
           (transport/send transport (misc/response-for msg resp))))
       print-var)))
 
+(defn- booleanize-bencode-val [m key]
+  (if (contains? m key)
+    ;; In bencode, empty list is treated as logical false.
```

Comment:
I think this is more of a convention that we've established (because of Elisp) than something common with Bencode in general. I'm not sure that CIDER or any client actually does this, though - I'm fairly certain we simply don't send the keys when something's false on the client-side. Anyways, I understand your desire to preserve the current behavior, I'm just commenting, because this part might be a bit misleading for someone.

Reply:
We could drop the current behavior if you are certain it's not used.

However, we have no way of representing `false` for cases when something is true by default, and the client wants to override that to false. Not sending the key then means preserving the default. I ran into this in cide-nrepl inspector middleware.

Alternatively, we can make it a convention to not use booleans in any APIs. That way, instead of `stream?` we would have something like `print-mode` that can be either "batch" or "stream".

Reply:
Well, that's fairly big topic and we can follow-up on it separately. I remembered I wrote down some thoughts here https://github.com/nrepl/nrepl/issues/196 a long time ago.